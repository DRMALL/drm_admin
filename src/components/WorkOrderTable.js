

import React from 'react'
import Table from 'antd/lib/table'
import { browserHistory } from 'react-router'
import TableTextHidden from './TableTextHidden'
import TableTime from './TableTime'
import getRichEditorValue from '../utils/getRichEditorValue'
import dispatch from '../actions/dispatch'
import workOrderDelById from '../actions/workOrderDelById'
import Modal from 'antd/lib/modal'
import delIcon from '../images/dels.png'


export default props => {
  const { workOrder, isShowModal } = props.state.workOrder
  console.log(workOrder)
  const columns = [{
    title:'标题',
    dataIndex: 'title',
    key: 'title',
    width: '15%',
    render: text => <div>
      {
        text.length > 40 ? <div>{text.slice(0,40)+ '...'}</div> : <div>{text}</div>
      }
    </div>
  },{
    title:'内容',
    dataIndex: 'content',
    key: 'content',
    width: '30%',
    render: text =>
    <div>
    {
      text[0]==='<' ? TableTextHidden( getRichEditorValue(text), 40 ) :
      TableTextHidden( text, 40 )
     }
    </div>
  },{
    title:'提问人',
    dataIndex: 'user.name',
    width: '10%',
    render: text =>  TableTextHidden( text, 10 )
  },{
    title:'状态',
    dataIndex: 'isHanlded',
    key: 'isHanlded',
    width: '10%',
  },{
    title:'用户反馈',
    dataIndex: 'isDone',
    key: 'isDone',
    width: '15%',
    render: text =>
      <div style={{ color: text==='已解决'?'#27b24a': '#d0011b' }} >{text}</div>
  },{
    title:'修改时间',
    dataIndex: 'updatedAtTime',
    key: 'updatedAtTime',
    width: '10%',
    render: text => TableTime(text, 15)
  },{
    title:'操作',
    dataIndex: 'action',
    width: '10%',
    render: (text, record) =>
      <div className='table-icon-content' >
        <div  data-order-id={ record._id }
             onClick={ navToOrderEdit } className='table-icon-item' >
          <i className="iconfont icon-compile" style={{ color: '#579df2', fontSize: 25 }} ></i>
        </div>
        <div className='table-icon-item' onClick={ delWorkOrderxx } data-work-order-id={ record._id }  >
          <img src={ delIcon } alt='删除' style={{ width:20,height:20 }} />
        </div>
      </div>
  },]
  return(
    <div>
      <Table columns={ columns } dataSource={ workOrder } rowKey={ record => record._id } />
      <Modal title='提示' visible={ isShowModal } okText='确定' cancelText='取消'
             onOk={ workOrderDelById } onCancel={ cancleDel }  >
        <p>确定删除此工单？</p>
      </Modal>
    </div>
    )
}

function delWorkOrderxx(e){
  const { workOrderId } = e.currentTarget.dataset
  dispatch('WORKORDER_GET_DEL_ORDER_ID', workOrderId)
}

function cancleDel(){
  dispatch('WORKORDER_CANCLE_ORDER')
}

function navToOrderEdit(e){
  const { orderId } = e.currentTarget.dataset
  localStorage.setItem('workOrderId', orderId)
  browserHistory.push('/workOrder/edit')
}
