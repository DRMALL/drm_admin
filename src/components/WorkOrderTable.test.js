

import React from 'react'
import WorkOrderTable from './WorkOrderTable'
import { render } from 'enzyme'
import toJson from 'enzyme-to-json'

it('WorkOrderTable test', () => {
  const state = { workOrder: { workOrder: [
    { title: '标题', content: 'asldjajtwe;ojawrw', user: 'admin', _id: '2323dsrw' },
    { title: '标题', content: 'asldjajtwe;ojawrw', user: 'admin', _id: '2323dsrds' }] } }
  const warpper = render(<WorkOrderTable state={ state } />)
  expect(toJson(warpper)).toMatchSnapshot()
})
