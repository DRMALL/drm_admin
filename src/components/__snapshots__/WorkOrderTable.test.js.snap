// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkOrderTable test 1`] = `
<div>
  <div
    class="ant-table-wrapper"
  >
    <div
      class="ant-spin-nested-loading"
    >
      <div
        class="ant-spin-container"
      >
        <div
          class="ant-table ant-table-large ant-table-scroll-position-left"
        >
          <div
            class="ant-table-content"
          >
            <div
              class="ant-table-body"
            >
              <table
                class=""
              >
                <colgroup>
                  <col
                    style="width:15%;min-width:15%;"
                  />
                  <col
                    style="width:30%;min-width:30%;"
                  />
                  <col
                    style="width:10%;min-width:10%;"
                  />
                  <col
                    style="width:10%;min-width:10%;"
                  />
                  <col
                    style="width:15%;min-width:15%;"
                  />
                  <col
                    style="width:10%;min-width:10%;"
                  />
                  <col
                    style="width:10%;min-width:10%;"
                  />
                </colgroup>
                <thead
                  class="ant-table-thead"
                >
                  <tr>
                    <th
                      class=""
                    >
                      <span>
                        标题
                      </span>
                    </th>
                    <th
                      class=""
                    >
                      <span>
                        内容
                      </span>
                    </th>
                    <th
                      class=""
                    >
                      <span>
                        提问人
                      </span>
                    </th>
                    <th
                      class=""
                    >
                      <span>
                        状态
                      </span>
                    </th>
                    <th
                      class=""
                    >
                      <span>
                        用户反馈
                      </span>
                    </th>
                    <th
                      class=""
                    >
                      <span>
                        修改时间
                      </span>
                    </th>
                    <th
                      class=""
                    >
                      <span>
                        操作
                      </span>
                    </th>
                  </tr>
                </thead>
                <tbody
                  class="ant-table-tbody"
                >
                  <tr
                    class="ant-table-row  ant-table-row-level-0"
                  >
                    <td
                      class=""
                    >
                      <span
                        class="ant-table-row-indent indent-level-0"
                        style="padding-left:0px;"
                      />
                      <div>
                        <div>
                          标题
                        </div>
                      </div>
                    </td>
                    <td
                      class=""
                    >
                      <div>
                        <div>
                          asldjajtwe;ojawrw
                        </div>
                      </div>
                    </td>
                    <td
                      class=""
                    />
                    <td
                      class=""
                    />
                    <td
                      class=""
                    >
                      <div
                        style="color:#d0011b;"
                      />
                    </td>
                    <td
                      class=""
                    />
                    <td
                      class=""
                    >
                      <div
                        class="table-icon-content"
                      >
                        <div
                          class="table-icon-item"
                          data-order-id="2323dsrw"
                        >
                          <i
                            class="iconfont icon-compile"
                            style="color:#579df2;font-size:25px;"
                          />
                        </div>
                        <div
                          class="table-icon-item"
                          data-work-order-id="2323dsrw"
                        >
                          <img
                            alt="删除"
                            src="dels.png"
                            style="width:20px;height:20px;"
                          />
                        </div>
                      </div>
                    </td>
                  </tr>
                  <tr
                    class="ant-table-row  ant-table-row-level-0"
                  >
                    <td
                      class=""
                    >
                      <span
                        class="ant-table-row-indent indent-level-0"
                        style="padding-left:0px;"
                      />
                      <div>
                        <div>
                          标题
                        </div>
                      </div>
                    </td>
                    <td
                      class=""
                    >
                      <div>
                        <div>
                          asldjajtwe;ojawrw
                        </div>
                      </div>
                    </td>
                    <td
                      class=""
                    />
                    <td
                      class=""
                    />
                    <td
                      class=""
                    >
                      <div
                        style="color:#d0011b;"
                      />
                    </td>
                    <td
                      class=""
                    />
                    <td
                      class=""
                    >
                      <div
                        class="table-icon-content"
                      >
                        <div
                          class="table-icon-item"
                          data-order-id="2323dsrds"
                        >
                          <i
                            class="iconfont icon-compile"
                            style="color:#579df2;font-size:25px;"
                          />
                        </div>
                        <div
                          class="table-icon-item"
                          data-work-order-id="2323dsrds"
                        >
                          <img
                            alt="删除"
                            src="dels.png"
                            style="width:20px;height:20px;"
                          />
                        </div>
                      </div>
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
        <ul
          class="ant-pagination ant-table-pagination"
          unselectable="unselectable"
        >
          <li
            aria-disabled="true"
            class="ant-pagination-disabled ant-pagination-prev"
            tabindex="0"
            title="上一页"
          >
            <a />
          </li>
          <li
            class="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
            tabindex="0"
            title="1"
          >
            <a>
              1
            </a>
          </li>
          <li
            aria-disabled="true"
            class="ant-pagination-disabled ant-pagination-next"
            tabindex="0"
            title="下一页"
          >
            <a />
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
